src/cluster_legacy.c

```diff
@@ -3045,7 +3045,13 @@ static void clusterProcessModulePacket(clusterMsgModule *module_data, clusterNod
     uint32_t len = ntohl(module_data->len);
     uint8_t type = module_data->type;
     unsigned char *payload = module_data->bulk_data;
-    moduleCallClusterReceivers(sender->name, module_id, type, payload, len);
+
+    /* Ensure sender name is properly truncated to 40 characters */
+    char truncated_name[41]; /* 40 chars + null terminator */
+    strncpy(truncated_name, sender->name, 40);
+    truncated_name[40] = '\0';
```

Comment:
Could you create SDS instead of manually copying and adding `\0` and send it in the callback and free it after that?